app/src/org/commcare/UpdateToPrompt.java

```diff
@@ -0,0 +1,158 @@
+            System.out.println(".apk version to prompt for update set to " + apkVersion);
+        } else {
+            System.out.println(".ccz version to prompt for update set to " + cczVersion);
+        }
+        if (this.forceByDate != null) {
+            System.out.println("force-by date is " + forceByDate);
+        }
+    }
+
+    public boolean isNewerThanCurrentVersion(CommCareApp currentApp) {
+        if (isApkUpdate) {
+            try {
+                Context c = CommCareApplication.instance();
+                PackageInfo pi = c.getPackageManager().getPackageInfo(c.getPackageName(), 0);
+                ApkVersion currentVersion = new ApkVersion(pi.versionName);
+                return currentVersion.compareTo(this.apkVersion) < 0;
+            } catch (PackageManager.NameNotFoundException e) {
+                System.out.println("Couldn't get current .apk version to compare with in " +
+                        "UpdateToPrompt: " + e.getMessage());
+                // This shouldn't happen, but it if it does, there's no way for us to know if the
+                // update version is newer, so don't prompt
+                return false;
+            }
+        } else {
+            int currentVersion = currentApp.getCommCarePlatform().getCurrentProfile().getVersion();
+            return currentVersion < this.cczVersion;
+        }
+    }
+
+    private void writeToPrefsObject(SharedPreferences prefs) {
+        ByteArrayOutputStream baos = null;
+        DataOutputStream serializedStream = null;
+        try {
+            baos = new ByteArrayOutputStream();
+            serializedStream = new DataOutputStream(baos);
+            try {
+                ExtUtil.write(serializedStream, this);
+                String serializedString = Base64.encodeToString(baos.toByteArray(), Base64.DEFAULT);
+                prefs.edit().putString(
+                        isApkUpdate ? KEY_APK_UPDATE_TO_PROMPT : KEY_CCZ_UPDATE_TO_PROMPT,
+                        serializedString).commit();
+                baos.close();
+                serializedStream.close();
+            } catch (IOException e) {
+                System.out.println("IO error encountered while serializing UpdateToPrompt: "
+                        + e.getMessage());
+            }
+        } finally {
+            try {
+                if (baos != null) {
+                    baos.close();
+                }
+                if (serializedStream != null) {
+                    serializedStream.close();
+                }
+            } catch (IOException e) {
+                // Nothing we can do
+            }
+        }
+    }
+
+    @Override
+    public void readExternal(DataInputStream in, PrototypeFactory pf) throws IOException, DeserializationException {
+        this.versionString = ExtUtil.readString(in);
+        this.isApkUpdate = ExtUtil.readBool(in);
+        this.hasForceByDate = ExtUtil.readBool(in);
+        if (hasForceByDate) {
```

Comment:
Oh, I guess just because I've never used that and wasn't super aware of it. Will adjust.